webapp/tests/karma/ts/effects/reports.effects.spec.ts

```diff
@@ -885,6 +908,56 @@ describe('Reports effects', () => {
       }]);
     }));
 
+    it('should allow selecting a different report while completing verification', fakeAsync(() => {
+      const selectedReports = [{
+        _id: 'report',
+        doc: {
+          _id: 'report',
+          _rev: 2,
+          contact: { _id: 'contact', name: 'name', parent: { _id: 'parent', type: 'clinic' } },
+        },
+      }];
+      authService.has.resolves(true);
+      store.overrideSelector(Selectors.getSelectedReports, selectedReports);
+      store.refreshState();
+
+      sinon.stub(Date, 'now').returns(1000); // using faketimers breaks fakeAsync's tick :(
+      // Updating the report causes it to be re-selected
+      dbService.put.callsFake(() => {
+        actions$ = of(ReportActionList.selectReport({id: 'report', silent: true}));
+      });
+      dbService.get.resolves({ _id: 'report', _rev: 3 });
```

Comment:
Can we spice up this test a bit? Let's change it so the `get` calls the `selectReport` to select another report. 
I think we should also spy on `setFirstSelectedReportDocProperty` and `setFirstSelectedReportFormattedProperty` to make sure that these actions don't end up overwriting any of the properties from the other selected report. 

This way the test becomes different to the one above.

Reply:
Okay, I added a third call to `selectReport`, triggered by the `dbService.get` call.  Then I also added asserts for `setFirstSelectedReportDocProperty` and `setFirstSelectedReportFormattedProperty` that basically just make sure that we only call each once with the expected data (technically there are two earlier calls to setFirstSelectedReportDocProperty in the workflow to set other data, but I just make sure that there is only one more after those that sets the expected `_rev` value.

Let me know if this was not what you were thinking.